File: Data Structures & Algorithms/depth-of-binary-tree/submission-3.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public int maxDepth(TreeNode root) {
        if(root == null)
        {
            return 0;
        }
        Queue<TreeNode> q = new ArrayDeque<>();
        q.add(root);
        int count = 0;
        while(!q.isEmpty())
        {
            int sz = q.size();
            count++;
            for(int i = 0; i < sz; i++)
            {
                TreeNode nd = q.remove();
                if(nd.left != null)
                {
                    q.add(nd.left);
                }
                if(nd.right != null)
                {
                    q.add(nd.right);
                }
            }
        }
        return count;
    }
}

File: Data Structures & Algorithms/is-anagram/submission-3.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        left = {}
        for x in s:
            if x in left:
                left[x] += 1
            else:
                left[x] = 1
        right = {}
        for x in t:
            if x in right:
                right[x] += 1
            else:
                right[x] = 1
        for x in left:
            if x not in right:
                return False
            if x in right and left[x] != right[x]:
                return False
        for x in right:
            if x not in left:
                return False
            if x in left and left[x] != right[x]:
                return False
        return True
                
        